component/loki/source/cloudflare/cloudflare.go

```diff
@@ -73,10 +75,19 @@ func (c *Arguments) Validate() error {
 	if c.PullRange < 0 {
 		return fmt.Errorf("pull_range must be a positive duration")
 	}
-	_, err := cft.Fields(cft.FieldsType(c.FieldsType))
+	_, err := cft.Fields(cft.FieldsType(c.FieldsType), c.CustomFields)
 	if err != nil {
-		return fmt.Errorf("invalid fields_type set; the available values are 'default', 'minimal', 'extended' and 'all'")
+		return fmt.Errorf("invalid fields_type set; the available values are 'default', 'minimal', 'extended', 'custom' and 'all'")
 	}
+
+	if cft.FieldsType(c.FieldsType) == cft.FieldsTypeAll && len(c.CustomFields) > 0 {
+		return fmt.Errorf("no need to provide custom fields when fields_type is set to 'all' as it contains already all available fields")
+	}
```

Comment:
Nit: I think we can remove this check, as it is not harmful to the user and we know how to do the right thing (use all fields). There may be other opinions on this though ;)